src/app/modules/bounce/bounce.controller.ts

```diff
@@ -23,11 +23,24 @@ export const handleSns: RequestHandler<
   // so we never fail on malformed input. The response code is meaningless since
   // it is meant to go back to AWS.
   try {
-    const isValid = await snsService.isValidSnsRequest(req.body)
-    if (!isValid) {
-      return res.sendStatus(StatusCodes.FORBIDDEN)
+    const isValid = await BounceService.isValidSnsRequest(req.body)
+    if (!isValid) return res.sendStatus(StatusCodes.FORBIDDEN)
+
+    const notification: IEmailNotification = JSON.parse(req.body.Message)
+    BounceService.logEmailNotification(notification)
+    const bounceDoc = await BounceService.getUpdatedBounceDoc(notification)
+    // Could not parse notification correctly
+    if (!bounceDoc) return res.sendStatus(StatusCodes.BAD_REQUEST)
+    // TODO (private #30): enable form deactivation
+    // if (bounceDoc.areAllPermanentBounces()) {
+    //   await BounceService.deactivateFormFromBounce(bounceDoc)
+    // }
+    if (bounceDoc.isCriticalBounce()) {
+      const emailRecipients = await BounceService.notifyAdminOfBounce(bounceDoc)
+      bounceDoc.updateHasEmailed(emailRecipients)
```

Comment:
nit: rename to `setHasEmailed`? update feels like the document will save, but that is not the case